lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,106 @@
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.open(go_sum_path).each("=\n") do |line|
+        dep_list.append({ "fullDependency" => line, "name" => get_name(line),
```

Comment:
Nit: For readability can we do something along the lines of
```
dep_list.append(
 { 
   "fullDependency" => line, 
   "name" => get_name(line), 
   "version" => get_version(line) 
 }
)

Reply:
Yep, changed 👍